### PR TITLE
Fix for adding pkcs9 contentType entry name

### PR DESCRIPTION
**Pull request by lealem47**
# Description

Fix for adding pkcs9 contentType entry name

Fixes zd#16348

# Testing

Using customers provided reproducer

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfssl/wolfcrypt/asn.h [hunk 1]
```diff
@@ -707,6 +707,8 @@ enum DN_Tags {
     ASN_DNQUALIFIER   = 0x2e,   /* dnQualifier */
 #endif /* WOLFSSL_CERT_NAME_ALL */
 
+    ASN_CONTENT_TYPE  = 0x03,   /* pkcs9_contentType */
```
Comment (embhorn) [inline wolfssl/wolfcrypt/asn.h:710]: This doesn't conflict with `ASN_COMMON_NAME   = 0x03,   /* CN */` ?

**Review by embhorn — CHANGES_REQUESTED** [review]
Also waiting for customer feedback on zd16348

File: wolfssl/wolfcrypt/asn.h (hunk 1 above)
Comment (lealem47) [reply wolfssl/wolfcrypt/asn.h:710]: I think both are `0x03`. Oid defined here https://datatracker.ietf.org/doc/html/rfc2630#section-11.1

**Comment by lealem47** [comment]
Retest this please

**Review by embhorn — APPROVED** [review]
On to final review / merge

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]
Looks like there is one test in api.c with NID_pkcs9_contentType. Please expand it to flag the issue that this is fixing.

File: wolfssl/wolfcrypt/asn.h (hunk 1 above)
Comment (lealem47) [reply wolfssl/wolfcrypt/asn.h]: This was spot on. Their identifier id's were the same but different base OID's. Fixed in the second commit.

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**